### PR TITLE
Add detailed CSV export option

### PR DESCRIPTION
**Pull request by talvasconcelos**
Optional detailed CSV export with the TPoS items list.

Normal behaviour:
```
"Status","Memo","Date","Amount (sats)","Fee (msats)","Tag","Payment Hash","Payment Proof","Webhook","Fiat Currency","Fiat Amount"
"success","Split payment: 10.0% for test split","2024-07-30 11:32","-162","0","splitpayments","fbd0dfdc19a53ec11adfb04773ebe560a9ed388b094de61a5ef2a5a853f23ea2","","","EUR","0.1"
"success","Split payment: 25.0% for wallet2","2024-07-30 11:32","-406","0","splitpayments","ba7537d0158dbe551c3e74c0ed0d3b5c2364d8a1df6df47c80b6ff4c8932a19c","","","EUR","0.25"
"success","€1.00 to normal","2024-07-30 11:31","1626","0","tpos","165d6b2c7e5bb1825d96c6866540b581080bd222ab3c95980a212eec110937a8","","","EUR","1.001"
"success","Split payment: 10.0% for test split","2024-07-30 11:29","-618","0","splitpayments","4cd9b47d126bddfbc18f1d3ca498b0932f99b79fccf349f5b1fc3c2e4b3803a9","","","EUR","0.38"
```

Detailed:
```
"Status","Memo","Date","Amount (sats)","Fee (msats)","Tag","Payment Hash","Payment Proof","Webhook","Fiat Currency","Fiat Amount","Items"
"success","Split payment: 10.0% for test split","2024-07-30 11:32","-162","0","splitpayments","fbd0dfdc19a53ec11adfb04773ebe560a9ed388b094de61a5ef2a5a853f23ea2","","","EUR","0.1",""
"success","Split payment: 25.0% for wallet2","2024-07-30 11:32","-406","0","splitpayments","ba7537d0158dbe551c3e74c0ed0d3b5c2364d8a1df6df47c80b6ff4c8932a19c","","","EUR","0.25",""
"success","€1.00 to normal","2024-07-30 11:31","1626","0","tpos","165d6b2c7e5bb1825d96c6866540b581080bd222ab3c95980a212eec110937a8","","","EUR","1.001",""
"success","Split payment: 10.0% for test split","2024-07-30 11:29","-618","0","splitpayments","4cd9b47d126bddfbc18f1d3ca498b0932f99b79fccf349f5b1fc3c2e4b3803a9","","","EUR","0.38",""
"success","Split payment: 25.0% for wallet2","2024-07-30 11:29","-1545","0","splitpayments","f0c886620b76fc1db730345c3605de3c8b96066fa67410128c6c95d4da09c785","","","EUR","0.95",""
"success","€3.80 to normal","2024-07-30 11:29","6183","0","tpos","88af00376b31b42049982366b8a828dcf7c2936beddaeef74678eba9436d3a25","","","EUR","3.801","{""currency"":""EUR"",""exchangeRate"":1627.013185264484,""items"":[{""price"":1,""formattedPrice"":""€1.00"",""quantity"":1,""title"":""test 1"",""tax"":10},{""price"":1.99,""formattedPrice"":""€1.99"",""quantity"":1,""title"":""test 2"",""tax"":23}],""taxIncluded"":false,""taxValue"":0.5577}"
"success","Split payment: 25.0% for wallet2","2024-07-30 11:24","-419","0","splitpayments","38c490645473c6608eaaa764de27c1ce464761d347b0c1ce420b0fc460037e4d","","","EUR","0.257",""
"success","Split payment: 10.0% for test split","2024-07-30 11:24","-167","0","splitpayments","90d2249aa557ad8f65b62a676779c28f58d01b0d969a3e06a61a7a1ad6b15fe0","","","EUR","0.103",""
```

Also fixed `pending` column on the paymentsCSV, it's now `status`

Closes #2325 , #2312

### CODE REVIEW

**Comment by talvasconcelos** [comment]
Fuck... this needs the bundle?! @dni ?

`make bundle` ?

**Comment by dni** [comment]
yep, components needs bundling :dagger: 

can you remove that 13 merge commits :D?

**Comment by talvasconcelos** [comment]
> yep, components needs bundling 🗡️
> 
> can you remove that 13 merge commits :D?

ser... I'm too stupid to know how to remove the commits

**Comment by motorina0** [comment]
Is this what one expects when exporting?

```csv
"Pending","Memo","Date","Amount (sats)","Fee (msats)","Tag","Payment Hash","Payment Proof","Webhook","Fiat Currency","Fiat Amount","Items"
"","$12.00 to t1","2024-07-31 12:15","-18150","0","","f2d8d157355094f5635ca0d7e9fd992e7a2f8c32a24db79575fd674663209445","","","","",""
"","Admin top up","2024-07-31 12:15","18150","0","tpos","f2d8d157355094f5635ca0d7e9fd992e7a2f8c32a24db79575fd674663209445","","","","","{""currency"":""USD"",""exchangeRate"":1512.4347447633495,""items"":[{""price"":10,""formattedPrice"":""$10.00"",""quantity"":1,""title"":""i1"",""tax"":20}],""taxIncluded"":false,""taxValue"":2}"
```

**Comment by motorina0** [comment]
NOK:
image

OK:
add `outline` to the `q-btn-dropdown`
image

File: lnbits/static/js/components/payment-list.js [hunk 1]
```diff
@@ -158,6 +158,14 @@ Vue.component('payment-list', {
       const params = new URLSearchParams(query)
       LNbits.api.getPayments(this.wallet, params).then(response => {
         const payments = response.data.data.map(LNbits.map.payment)
+        if (detailed) {
+          this.paymentsCSV.columns.push({
+            name: 'items',
+            align: 'right',
+            label: 'Items',
+            field: row => (row.tag === 'tpos' && row.extra.details) || null
```
Comment (motorina0) [inline lnbits/static/js/components/payment-list.js:166]: why only `tpos`?

Comment (talvasconcelos) [reply lnbits/static/js/components/payment-list.js:166]: only tpos has the details `extra` and it's the only extension until now to have item

**Comment by talvasconcelos** [comment]
> Is this what one expects when exporting?
> 
> ```
> {""currency"":""USD"",""exchangeRate"":1512.4347447633495,""items"":[{""price"":10,""formattedPrice"":""$10.00"",""quantity"":1,""title"":""i1"",""tax"":20}],""taxIncluded"":false,""taxValue"":2}"
> ```

yes, it's JSON stringify of the items, the user can use it to build whatever type of report needed, i guess

File: lnbits/static/js/components/payment-list.js (hunk 1 above)
Comment (motorina0) [reply lnbits/static/js/components/payment-list.js:166]: 1. why exclude other/future ones?
2. core should not know/care about a particular extension

Comment (talvasconcelos) [reply lnbits/static/js/components/payment-list.js:166]: should be just `field: row => (row.extra?.details) || null` ?

Comment (motorina0) [reply lnbits/static/js/components/payment-list.js:166]: or simply `field: row => row.extra?.details)` ?

**Comment by motorina0** [comment]
https://github.com/user-attachments/assets/143a5850-2f02-4ceb-9934-f822b50666bb

File: lnbits/static/js/components/payment-list.js [hunk 2]
```diff
@@ -33,13 +33,15 @@ Vue.component('payment-list', {
         search: null,
         loading: false
       },
+      exportTagName: '',
+      exportPaymentTagList: [],
       paymentsCSV: {
         columns: [
           {
-            name: 'pending',
+            name: 'status',
```
Comment (motorina0) [inline lnbits/static/js/components/payment-list.js:41]: hmmm, this was not part of the PR

**Comment by dni** [comment]
hitting enter does not add the tag, i think it should

**Review by motorina0 — APPROVED** [review]

**Review by dni — APPROVED** [review]
